diff --git a/EfiLib/gnuefi-helper.c b/EfiLib/gnuefi-helper.c
index 8730403..032c7c7 100644
--- a/EfiLib/gnuefi-helper.c
+++ b/EfiLib/gnuefi-helper.c
@@ -109,6 +109,8 @@ UnicodeStrToAsciiStr (
   @return The length of String.
 
 **/
+// prevent duplicate definitions when compiled against gnu-efi 4.x
+#ifndef _GNU_EFI_4_0
 UINTN
 AsciiStrLen (
   IN      CONST CHAR8               *String
@@ -129,6 +131,7 @@ AsciiStrLen (
   }
   return Length;
 }
+#endif
 
 /**
   Determine whether a given device path is valid.
diff --git a/EfiLib/gnuefi-helper.h b/EfiLib/gnuefi-helper.h
index b691155..d3b4112 100644
--- a/EfiLib/gnuefi-helper.h
+++ b/EfiLib/gnuefi-helper.h
@@ -38,10 +38,12 @@ UnicodeStrToAsciiStr (
    OUT      CHAR8               *Destination
 );
 
+#ifndef _GNU_EFI_4_0
 UINTN
 AsciiStrLen (
    IN      CONST CHAR8               *String
 );
+#endif
 
 UINTN
 EFIAPI
diff --git a/EfiLib/legacy.c b/EfiLib/legacy.c
index 4ee6229..b1aeda0 100644
--- a/EfiLib/legacy.c
+++ b/EfiLib/legacy.c
@@ -16,6 +16,7 @@
  */
 
 #ifdef __MAKEWITH_GNUEFI
+#define GNU_EFI_USE_REALLOCATEPOOL_ABI 0
 #include "efi.h"
 #include "efilib.h"
 #include "gnuefi-helper.h"
diff --git a/filesystems/fsw_efi.h b/filesystems/fsw_efi.h
index bb51073..6182a46 100644
--- a/filesystems/fsw_efi.h
+++ b/filesystems/fsw_efi.h
@@ -42,8 +42,11 @@
 #include "../include/refit_call_wrapper.h"
 
 #ifdef __MAKEWITH_GNUEFI
+// prevent duplicate definitions when compiled against gnu-efi 4.x
+#ifndef _GNU_EFI_4_0
 #define CompareGuid(a, b) CompareGuid(a, b)==0
 #endif
+#endif
 
 #define REFIND_EFI_DISK_IO_PROTOCOL_GUID \
   { \
diff --git a/refind/lib.c b/refind/lib.c
index 7a9f28e..3fc20ef 100644
--- a/refind/lib.c
+++ b/refind/lib.c
@@ -56,6 +56,7 @@
  */
 
 #pragma pack(0)
+#define GNU_EFI_USE_REALLOCATEPOOL_ABI 0
 #include "global.h"
 #include "lib.h"
 #include "icns.h"
