--- a/config.m4		2024-12-17 20:20:06.141657014 +0100
+++ b/config.m4		2024-12-18 19:17:39.534963741 +0100
@@ -23,6 +23,17 @@
   src/cnn_face_detection.cc
   src/vector.cc"
 
+PHP_ARG_ENABLE(gif, whether to enable gif support,
+[  --enable-gif         Enable GIF image support])
+
+if test "$PHP_GIF" = "yes"; then
+  AC_CHECK_LIB([gif], [DGifOpenFileName], [],
+  [
+    AC_MSG_ERROR([libgif not found])
+  ])
+  PHP_ADD_LIBRARY_WITH_PATH([gif], [/usr/lib64], [GIF])
+fi
+
   AC_MSG_CHECKING(for pkg-config)
   AC_PATH_PROG(PKG_CONFIG, pkg-config, no)
   if test -x "$PKG_CONFIG"; then
@@ -65,4 +76,4 @@
   echo "Please submit bug reports at:"
   echo "  https://github.com/goodspb/pdlib/issues"
   echo ""
-])
\ No newline at end of file
+])
